contrib/buildinfo/src/mill/contrib/buildinfo/BuildInfo.scala

```diff
@@ -1,58 +1,200 @@
 package mill.contrib.buildinfo
 
-import mill.T
-import mill.Agg
-import mill.api.{Logger, Loose, PathRef}
-import mill.define.Target
-import mill.scalalib.{Dep, ScalaModule, DepSyntax}
+import mill.{PathRef, T}
+import mill.scalalib.{JavaModule, ScalaModule}
 
-trait BuildInfo extends ScalaModule {
-
-  def buildInfoPackageName: Option[String] = None
+trait BuildInfo extends JavaModule {
+  /**
+   * The package name under which the BuildInfo data object will be stored.
+   */
+  def buildInfoPackageName: String
 
+  /**
+   * The name of the BuildInfo data object, defaults to "BuildInfo"
+   */
   def buildInfoObjectName: String = "BuildInfo"
 
-  def buildInfoMembers: T[Map[String, String]] = T {
-    Map.empty[String, String]
+  /**
+   * Enable to compile the BuildInfo values directly into the classfiles,
+   * rather than the default behavior of storing them as a JVM resource. Needed
+   * to use BuildInfo on Scala.js which does not support JVM resources
+   */
+  def buildInfoStaticCompiled: Boolean = false
+
+  /**
+   * A mapping of key-value pairs to pass from the Build script to the
+   * application code at runtime.
+   */
+  def buildInfoMembers: T[Seq[BuildInfo.Value]]
+
+  def resources =
```

Comment:
The file generation should be a dedicated target. This helps when someone needs to override the `resources` target but wants to keep some overridden functionality, or if the file is needed elsewhere.

Reply:
done